src/Presentation/Nop.Web/Global.asax.cs

```diff
@@ -46,6 +47,9 @@ protected void Application_Start()
             //disable "X-AspNetMvc-Version" header name
             MvcHandler.DisableMvcResponseHeader = true;
 
+            //MiniProfiler for EF must be initialized before first Database call. Doesn't matter if we later start Profiler or not.
+            MiniProfilerEF6.Initialize();

```

Comment:
That's bad. In this case MiniProfiler will always be used. Even in production environments where it's not required. Hence it can affect performance